### PR TITLE
Refactor Edit Phrases

### LINKED ISSUE #296: Refactor: Edit Phrases

### PR DESCRIPTION
**Pull request by stevefosterwta**
Closes github issue: 296

# Description of Work
Refactors `EditPhrasesViewController`

## Notes to Test
The changes in this PR assumes the following feature flag configuration:
- `EditPhrasesEnabled` is not enabled 
- `RefactoredInterfaceEnabled` is enabled

### CODE REVIEW

**Review by Clstroud — APPROVED** [review]
💯

File: Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift [hunk 1]
```diff
@@ -122,6 +85,7 @@ final class EditPhrasesViewController: UIViewController, UICollectionViewDelegat
         case (.compact, .regular):
             collectionView.layout.numberOfColumns = .fixedCount(1)
             collectionView.layout.numberOfRows = .fixedCount(3)
+            collectionView.layout.numberOfRows = .minimumHeight(130)
```
Comment (stevefosterwta) [inline Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift]: I don't think this belongs here.

File: Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift [hunk 2]
```diff
@@ -180,6 +144,33 @@ final class EditPhrasesViewController: UIViewController, UICollectionViewDelegat
         collectionView.backgroundView = nil
     }
 
+    // MARK: Actions 
+
+    @IBAction private func addPhrasePressed() {
+        let storyboard = UIStoryboard(name: "EditTextViewController", bundle: nil)
+        let vc = storyboard.instantiateViewController(identifier: "EditTextViewController") as! EditTextViewController
```
Comment (stevefosterwta) [inline Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift]: This should cause a crash.

File: Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
-final class EditPhrasesViewController: UIViewController, UICollectionViewDelegate, NSFetchedResultsControllerDelegate {
+final class EditPhrasesViewController: PagingCarouselViewController, NSFetchedResultsControllerDelegate {
 
     var category: Category!
 
-    private var carouselCollectionViewController: CarouselGridCollectionViewController?
-    private var disposables = Set<AnyCancellable>()
-
-    private lazy var diffableDataSource = CarouselCollectionViewDataSourceProxy<Int, PhraseViewModel>(collectionView: collectionView!) { [weak self] (collectionView, indexPath, phrase) -> UICollectionViewCell? in
+    private lazy var diffableDataSource = CarouselCollectionViewDataSourceProxy<Int, PhraseViewModel>(collectionView: collectionView) { [weak self] (collectionView, indexPath, phrase) -> UICollectionViewCell? in
```
Comment (Clstroud) [inline Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift:17]: Can we add a warning as a reminder to circle back and modernize this to use the diffable datasource snapshot NSFRCDelegate strategy? We shouldn't need the PhraseViewModel anymore with that strategy

File: Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift (hunk 2 above)
Comment (Clstroud) [inline Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift]: I think this may crash with the latest version of `EditTextViewController`

**Review by Clstroud — APPROVED** [review]
Two suggestions, but overall 👍

File: Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift (hunk 2 above)
Comment (stevefosterwta) [reply Vocable/Features/Settings/EditPhrases/EditPhrasesViewController.swift]: Fixed, again.

**Comment by stechkev** [comment]
Issue #1:
Buttons have a dark overlay when highlighted. 
The following buttons, when highlighted, have all the proper animations except that they dim when they are being hovered or pressed:
- Timing and sensitivity page
  - "-", "+", "low", "medium", "high"
- Categories
  - "^", "V", ">" at the end of the category
- My Sayings
  - "pencil icon", "trash icon"
- Custom category
  - "pencil icon"

**Comment by stechkev** [comment]
Issue #2: 
"Remove category" button on default categories should display in a disabled state

**Comment by stechkev** [comment]
issue #3:
"X" and "<-" buttons in upper left hand corner of certain screens don't have their left edge lined up with the buttons underneath.
screens this affects:
- Categories
- Settings
- Timing and sensitivity

**Comment by stechkev** [comment]
There are fixes to categories/category detail that will come in a separate PR.
Same with Selection Mode and Timing Sensitivity.

**Pull request merged**